qt/scientific_interfaces/test/ISISReflectometry/Reduction/ParseReflectometryStringsTest.h

```diff
@@ -140,6 +140,17 @@ class ParseReflectometryStringsTest : public CxxTest::TestSuite {
     TS_ASSERT(!result.is_initialized());
   }
 
+  void testParseScaleFactorRejectsZero() {
+    auto result = parseScaleFactor("0.0");
+    TS_ASSERT(!result.is_initialized());
```

Comment:
There seems to be some inconsistency within this file, and I am potentially misunderstanding the use of `.get()`. On line 117 there appears to be a similar assertion `TS_ASSERT(!result.get().is_initialized());` - how do these differ?

Reply:
It's a bit of an unintuitive interface but looking at ParseReflectometryStringsTest.h:104 you can see invalid options return a boost:none (or !is_initialized() )

Whereas, on line 132 of the same file whitepsace isn't incorrect (so the first .get works) but there isn't anything there, so the second layer is now boost:none